src/pager.c

```diff
@@ -697,8 +697,9 @@ struct Pager {
   char *pTmpSpace;            /* Pager.pageSize bytes of space for tmp use */
   PCache *pPCache;            /* Pointer to page cache object */
 #ifndef SQLITE_OMIT_WAL
-  Wal *pWal;                  /* Write-ahead log used by "journal_mode=wal" */
-  char *zWal;                 /* File name for write-ahead log */
+  Wal *pWal;                       /* Write-ahead log used by "journal_mode=wal" */
+  libsql_wal_methods *pWalMethods; /* Virtual methods for interacting with WAL */
+  char *zWal;                      /* File name for write-ahead log */
```

Comment:
Would it be possible to push it down to libsql_wal_methods somehow to make the whole upper layer not aware of what file WAL is using? I'm asking because I can imagine a situation when there's more than one WAL file or no files at all.

Reply:
It seems that zWal is only passed to:
pWalMethods->xOpen
sqlite3OsAccess
sqlite3OsDelete

Reply:
Sounds possible, but since it was there in the original implementation, let's leave it for a separate PR

Reply:
actually no, you're right, let's get rid of this, this parameter is useless for a general purpose WAL, it's an implementation detail of the original implementation

Reply:
done!